### PR TITLE
Fix sesolve tests

### PR DESCRIPTION
**Pull request by pierreguilmin**
Fixes correctness and gradient test for euler and adaptive (for the cavity system).

### CODE REVIEW

File: tests/solver_tester.py [hunk 1]
```diff
@@ -11,227 +11,79 @@
...
+        true_Esave = system.expects(system.tsave)
         logging.warning(f'true_Esave = {true_Esave}')
-        assert jnp.allclose(result.Esave, true_Esave, rtol=esave_rtol, atol=esave_atol)
+        logging.warning(f'Esave      = {result.Esave}')
+        assert jnp.allclose(true_Esave, result.Esave, rtol=esave_rtol, atol=esave_atol)
 
     def test_correctness(self):
         pass
 
-    def _test_jit(
```
Comment (gautierronan) [inline tests/solver_tester.py:140]: Why remove the jit test?

**Review by gautierronan — APPROVED** [review]
LGTM. Just one question below. Also, I'll look into the tdqubit tests asap.

File: tests/solver_tester.py (hunk 1 above)
Comment (pierreguilmin) [reply tests/solver_tester.py:140]: I don't see the point of this test? If it's to test that the code works with JIT, it's jitted by default so already tested by the rest. If it's to test wether the jitted code runs faster, the answer will always be yes (I don't see a bug/situation where it's not the case). Moreover the no-jit code takes quite long to run (on the order of a few seconds), so once we test all the solvers it would slow down the CI quite a lot.
